app/javascript/components/molecules/Popover.js

```diff
@@ -47,20 +46,32 @@ const Popover = ({ buttonElement, children }) => {
 
   return (
     <>
-      {buttonElement({ ref: buttonRef, onClick: () => setShowPopper(!showPopper) })}
+      {buttonElement({
+        ref: setReferenceElement,
+        onClick: () => {
+          setShowPopper(!showPopper)
+          onButtonClick && onButtonClick()
+        }
+      })}
       {showPopper && (
         <div
           className='popper-container'
-          ref={popperRef}
+          ref={setPopperElement}
           style={styles.popper}
           {...attributes.popper}
         >
-          <div ref={setArrowRef} id='arrow' />
+          <div ref={setArrowElement} style={styles.arrow} id='arrow' />
           {children}
         </div>
       )}
     </>
   )
 }
 
+Popover.propTypes = {
+  buttonElement: PropTypes.func.isRequired,
+  children: PropTypes.element,
```

Comment:
I think you want this to be PropTypes.node

I think the difference is node is anything that can be rendered (string, number, element) but element is just react components.